flash/text/seq2seq/core/data.py

```diff
@@ -222,6 +223,62 @@ def __setstate__(self, state):
         self.tokenizer = AutoTokenizer.from_pretrained(self.backbone, use_fast=True)
 
 
+class Seq2SeqDictionaryDataSource(Seq2SeqDataSource):
+
+    def _tokenize_fn(
+        self,
+        example: Dict[str, str],
+        input: Optional[str] = 'input',
+        input_pair: Optional[str] = None,
+        target: Optional[str] = None,
+    ) -> Callable:
+
+        ex_input = example[input]
+        ex_input_pair = example[input_pair] if input_pair else None
+        ex_target = example[target] if target else None
+
+        model_inputs = self.tokenizer(
+            ex_input, ex_input_pair, max_length=self.max_source_length, padding=self.padding, truncation=True
+        )
+
+        # Setup the tokenizer for targets
+        if ex_target is not None:
+            with self.tokenizer.as_target_tokenizer():
+                labels = self.tokenizer(
+                    ex_target, max_length=self.max_target_length, padding=self.padding, truncation=True
+                )
+
+            model_inputs["labels"] = labels["input_ids"]
+
+        return model_inputs
+
+    def load_data(self, data: Any, columns: List[str] = None) -> 'datasets.Dataset':
+        if columns is None:
+            columns = ["input_ids", "attention_mask", "labels"]
+            if self._running_stage.value == RunningStage.PREDICTING:
+                columns.remove("labels")
+
+        stage = self._running_stage.value
+        data, input, input_pair, target = data
```

Comment:
Let's just enforce `input="question"`, `input_pair="context"` and `target=None` (i.e. this will only be used for prediction). That way users can just pass the dictionary to the predict function rather than the tuple.